balancer/pickfirst/pickfirstleaf/pickfirstleaf.go

```diff
@@ -153,14 +170,18 @@ type pickfirstBalancer struct {
 	// The mutex is used to ensure synchronization of updates triggered
 	// from the idle picker and the already serialized resolver,
 	// SubConn state updates.
-	mu    sync.Mutex
-	state connectivity.State
+	mu                sync.Mutex
+	connectivityState connectivity.State
```

Comment:
I think this deserves a comment, too, since there are now two very similar fields.

Reply:
Shouldn't this kind of tracking be done in the subconn struct (`scData`) and not here?  I would expect the lb policy only has the `concludedState` and each subchannel needs to track its real state and its effective state, accounting for sticky-TF and health reporting?  It seems confusing to me that the LB policy itself is tracking two different states, but I'm willing to believe it's simpler this way if you tried it the other way already.

Reply:
I was referring the Java implementation of Pickfirst and they were handling the sticky TF behaviour in the LB Policy. I don't see any issue with handling sticky TF in the subchannel state. I've update the PR to reflect the suggestions.